src/app/cloud/settings/client_page.tsx

```diff
@@ -78,6 +129,38 @@ export const SettingsPage = () => {
     [updateUser],
   )
 
+  const deleteAccount = React.useCallback(async ({ data }) => {
+    if (user) {
+      try {
+        const confirmedUser = await login({
```

Comment:
Why do we need to login here?

Reply:
Oh I see, you're having them log in to confirm the action via the modal, ok 👍

Reply:
One problem I did find here actually is that I technically could log in as a different user. This would change my token and break my session. We need to validate that the user they have entered matches their currently authenticated user before attempting login.

Reply:
Ok I've got this fixed up in https://github.com/payloadcms/website/pull/147/commits/6613d3ab980d88bfaeee1d472a867085198b67b7